### PR TITLE
Prepare 23.07 docker images

### PR DESCRIPTION
**Pull request by mathias-arm**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -45,3 +45,22 @@ sdk/target
 */.idea/
 */*/.idea/
 build/**
+
+execution-engine/Cargo.lock
```
Comment (egrimley-arm) [inline .gitignore:49]: Any reason not to use `*/Cargo.lock` here?

File: docker/Makefile [hunk 2]
```diff
@@ -20,9 +20,16 @@ DOCKER_GROUP_ID=$(shell stat --format "%g" $(shell readlink -f /var/run/docker.s
 NE_GID := $(shell getent group ne | awk -F: '{printf $$3}')
 endif
 
+ifeq ($(ARCH),x86_64)
+DOCKER_ARCH = amd64/
```
Comment (egrimley-arm) [inline docker/Makefile:24]: The trailing `/` is a bit unexpected. Is it necessary?

File: docker/Makefile [hunk 3]
```diff
@@ -50,6 +57,10 @@ base: base/Dockerfile
 all-base: base linux-base nitro-base icecap-base
 	echo 'All base docker images re-built from scratch'
 
+# Make sure the cargo registry directory exists to avoid permission issues
+$(HOME)/.cargo/registry:
+	mkdir -p $(HOME)/.cargo/registry
```
Comment (egrimley-arm) [inline docker/Makefile:62]: You can use `$@` instead of repeating the target.

File: transport-protocol/.gitignore [hunk 4]
```diff
@@ -1 +1,2 @@
+src/transport_protocol.rs
```
Comment (egrimley-arm) [inline transport-protocol/.gitignore:1]: I think this file is no longer created so it shouldn't be necessary to ignore it.

File: .gitignore (hunk 1 above)
Comment (mathias-arm) [reply .gitignore:49]: I wanted to be sure that I would not ignore more files than I needed to but your suggestion is more future-proof.

File: docker/Makefile (hunk 2 above)
Comment (mathias-arm) [reply docker/Makefile:24]: Yes. The Dockerfile does not support the case where the `/` is not in the variable.

**Review by egrimley-arm — APPROVED** [review]

**Pull request merged**